### PR TITLE
Provider client specific alerting tests

### PR DESCRIPTION
**Pull request by fbalak**

### CODE REVIEW

File: tests/functional/monitoring/conftest.py [hunk 1]
```diff
@@ -1133,3 +1133,76 @@ def teardown():
...
+
+    current_version = storageconsumer.get_ocs_version()
+    logger.info(f"Reported client version: {current_version}")
+    original_cluster = config.cluster_ctx
+    logger.info(f"Provider cluster key: {original_cluster}")
+    logger.info("Switch to client cluster")
+    config.switch_to_consumer()
+    client_cluster = config.cluster_ctx
+    logger.info(f"Client cluster key: {client_cluster}")
+    cluster_id = run_cmd(
```
Comment (DanielOsypenko) [inline tests/functional/monitoring/conftest.py:1162]: run_cmd is the deprecated form of exec_cmd; not sure if affects. Please leave or fix by your decision.

File: tests/functional/monitoring/conftest.py [hunk 2]
```diff
@@ -1133,3 +1133,76 @@ def teardown():
...
+        Stop heartbeat and change value of ocs version in storage client resource
+        for 3 minutes.
+
+        """
+        nonlocal client
+        nonlocal original_cluster
+        # run_time of operation
+        run_time = 60 * 3
+        client.stop_heartbeat()
+        client.change_ocs_version()
```
Comment (DanielOsypenko) [inline tests/functional/monitoring/conftest.py:1179]: can't find **change_ocs_version** method. Maybe missed something, but class StorageConsumer is a newly added class and the closest it has is **set_ocs_version**

File: ocs_ci/ocs/resources/storageconsumer.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        self.consumer_context = consumer_context
+        self.name = consumer_name
+        self.ocp = ocp.OCP(
+            resource_name=self.name,
+            kind=constants.STORAGECONSUMER,
+            namespace=config.cluster_ctx.ENV_DATA["cluster_namespace"],
+        )
+        if self.consumer_context:
+            self.heartbeat_cronjob = self.get_heartbeat_cronjob()
+            self.original_context = config.cluster_ctx
```
Comment (DanielOsypenko) [inline ocs_ci/ocs/resources/storageconsumer.py:35]: maybe better to name self.original_context as provider_context, that should go well with existing self.consumer_context

File: ocs_ci/ocs/resources/storageconsumer.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            namespace=config.cluster_ctx.ENV_DATA["cluster_namespace"],
+        )
+        cronjob = [
+            ocp.OCP(**job)
+            for job in cronjobs_obj.get().get("items")
+            if job.name.endswith("status-reporter")
+        ][0]
+        return cronjob
+
+    def _switch_original_cluster(self):
```
Comment (DanielOsypenko) [inline ocs_ci/ocs/resources/storageconsumer.py:108]: Suggestion to rename original cluster to provider_cluster, or storage_cluster. WDYT?

**Comment by fbalak** [comment]
Verified: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/2259/console
The reproduced state is generated correctly but the test failed on different msg in an alert. In the broken test setup is raised multiple alerts of the same type with different messages containing different names of clusters. The correct message is included but there are also incorrect alerts raised due to problems with the setup with different consumer clusters that prevent from the test to pass. With valid clusters this should pass.

**Review by DanielOsypenko — APPROVED** [review]

**Review by dahorak — APPROVED** [review]
Just one question, otherwise LGTM.

File: ocs_ci/ocs/ocp.py [hunk 5]
```diff
@@ -421,13 +421,14 @@ def apply(self, yaml_file):
         command = f"apply -f {yaml_file}"
         return self.exec_oc_cmd(command)
 
-    def patch(self, resource_name="", params=None, format_type=""):
+    def patch(self, resource_name="", params=None, subresource=None, format_type=""):
```
[the remark this answers is not in the document]
Comment (fbalak) [reply ocs_ci/ocs/ocp.py:424]: The change was not used after all. I will drop it.

**Review by dahorak — APPROVED** [review]

**Review by DanielOsypenko — APPROVED** [review]

**Pull request merged**